pkg/bufferdecoder/eventsreader_test.go

```diff
@@ -188,6 +188,88 @@ func TestReadArgFromBuff(t *testing.T) {
 	}
 }
 
+func TestReadStringVarFromBuff(t *testing.T) {
+	tests := []struct {
+		name           string
+		buffer         []byte
+		max            int
+		expected       string
+		expectedCursor int
+		expectError    bool
+	}{
+		{
+			name:           "Null terminated string in larger buffer",
+			buffer:         []byte{'H', 'e', 'l', 'l', 'o', 0, 'W', 'o', 'r', 'l', 'd'},
+			max:            6,
+			expected:       "Hello",
+			expectedCursor: 6,
+			expectError:    false,
+		},
+		{
+			name:           "Buffer with same length as max without null terminator",
+			buffer:         []byte{'H', 'e', 'l', 'l', 'o'},
+			max:            5,
+			expected:       "Hell",
+			expectedCursor: 5,
+			expectError:    false,
+		},
+		{
+			name:           "Buffer longer than max length without null terminator",
+			buffer:         []byte{'H', 'e', 'l', 'l', 'o', 'W', 'o', 'r', 'l', 'd'},
+			max:            5,
+			expected:       "Hell",
+			expectedCursor: 5,
+			expectError:    false,
+		},
+		{
+			name:           "Zero max length",
+			buffer:         []byte{'H', 'e', 'l', 'l', 'o', 0, 'W', 'o', 'r', 'l', 'd'},
+			max:            0,
+			expected:       "",
+			expectedCursor: 0,
+			expectError:    false,
+		},
```

Comment:
If we set max 0, an empty string should be expected? Or should the helper disregard the max and check the buffer until finding a nul char?